geo/src/main/java/org/odk/collect/geo/selection/SelectionMapFragment.kt

```diff
@@ -209,14 +205,37 @@ class SelectionMapFragment(
         )
     }
 
-    private fun restoreZoomFromPreviousState(state: Bundle) {
+    private fun initializeViewport() {
+        previousState?.let {
+            if (restoreZoomFromPreviousState(it)) {
+                viewportInitialized = true
+            }
+        }
+
+        if (!viewportInitialized) {
+            itemsByFeatureId.filter { it.value.selected }.map { it.key }.firstOrNull()?.let {
+                onFeatureClicked(it)
+                viewportInitialized = true
+            }
+        }
+
+        if (!viewportInitialized && zoomToFeatureBoundingBoxOnLoad && points.isNotEmpty()) {
+            map.zoomToBoundingBox(points, 0.8, false)
+            viewportInitialized = true
+        }
+
+        map.setGpsLocationListener { point -> onLocationChanged(point) }
+    }
+
+    private fun restoreZoomFromPreviousState(state: Bundle): Boolean {
         val mapCenter: MapPoint? = state.getParcelable(MAP_CENTER_KEY)
         val mapZoom = state.getDouble(MAP_ZOOM_KEY)
 
         if (mapCenter != null) {
             map.zoomToPoint(mapCenter, mapZoom, false)
-            viewportInitialized = true // avoid recentering as soon as location is received
+            return true
```

Comment:
It made me uneasy to have this state change buried in here. Indicating that initialization is complete felt like it should go with all the different ways the viewport could be initialized.